lib/install.js

```diff
@@ -93,8 +93,14 @@ exports.install = function(targets, options) {
     if (options.link || options.quick)
       options.lock = true;
 
+    var defaultTargets = options.dev ? config.pjson.devDependencies : config.pjson.dependencies;
     if (targets === true)
-      targets = options.dev ? config.pjson.devDependencies : config.pjson.dependencies;
+      targets = defaultTargets;
+    else if (options.doUpdate)
+      Object.keys(defaultTargets).forEach(function(module) {
+        if (module in targets)
```

Comment:
@guybedford looking at this again, we should probably be checking that `targets[module]` is an empty constraint.

```
jspm install jquery@1.7
jspm update jquery@1.8 # does not update
```